apps/api/src/server/applications/application/documents/document.service.js

```diff
@@ -217,80 +225,63 @@ const upsertDocumentVersionsMetadataToDatabase = async (
+ * @param {DocumentToSaveExtended[]} documentsToUpload
  * @param {number} caseId
- * @returns {Promise<{response: {blobStorageHost: string, privateBlobContainer: string, documents: {blobStoreUrl: string, caseType: string, caseReference: string,documentName: string, GUID: string}[]} | null, failedDocuments: string[]}>}}
+ * @returns {Promise<{response: DocumentAndBlobInfoManyResponse | null, failedDocuments: string[]}>}}
  */
 export const obtainURLsForDocuments = async (documentsToUpload, caseId) => {
 	// Step 1: Retrieve the case object associated with the provided caseId
 	logger.info(`Retrieving case for caseId ${caseId}...`);
-
 	const caseForDocuments = await caseRepository.getById(Number(caseId), {});
-
 	logger.info(`Case retrieved: ${JSON.stringify(caseForDocuments)}`);
 
 	// Step 2: Check if the case object is found and has a reference
 	logger.info(`Checking if case has reference...`);
-
 	if (caseForDocuments == null || caseForDocuments.reference == null) {
 		throw new Error('Case not found or has no reference');
 	}
-
 	logger.info(`Case has reference`);
 
 	// Step 3: Map documents to the format expected by the database
 	logger.info(`Mapping documents to database format...`);
-
 	const documentsToSendToDatabase = mapDocumentsToSendToDatabase(caseId, documentsToUpload);
-
 	//throw Error(JSON.stringify(documentsToSendToDatabase));
-
 	logger.info(`Documents mapped: ${JSON.stringify(documentsToSendToDatabase)}`);
 
 	// Step 4: Add documents to the database if all are new
 	logger.info(`Attempting to insert documents to database...`);
-
 	const { successful, failed } = await attemptInsertDocuments(caseId, documentsToSendToDatabase);
 	if (successful.length === 0) {
 		logger.info(`Return early because all files failed to upload.`);
 		return { response: null, failedDocuments: failed };
 	}
-
 	logger.info(`Documents inserted: ${JSON.stringify(successful)}`);
 
 	// Step 5: Map documents to the format expected by the blob storage service
 	logger.info(`Mapping documents to blob storage format...`);
-
 	const requestToDocumentStorage = mapDocumentsToSendToBlobStorage(
 		successful,
 		caseForDocuments.reference
 	);
-
 	logger.info(`Documents mapped: ${JSON.stringify(requestToDocumentStorage)}`);
 
 	// Step 6: Send a request to the blob storage service to get the storage location for each document
-
 	logger.info(`Sending request to blob storage service...`);
-
 	const responseFromDocumentStorage = await getStorageLocation(requestToDocumentStorage);
-
 	logger.info(`Response from blob storage service: ${JSON.stringify(responseFromDocumentStorage)}`);
 
 	// Step 7: Upsert document versions metadata to the database
 	logger.info(`Upserting document versions metadata to database...`);
-
 	await upsertDocumentVersionsMetadataToDatabase(
 		responseFromDocumentStorage.documents,
 		responseFromDocumentStorage.privateBlobContainer
 	);
 
-	// @ts-ignore
-	/**
-	 * @type {any[]}
-	 */
+	/** @type {DocumentAndBlobInfoManyResponse[]} */
```

Comment:
This isn't an array of responses, but just an array of promises for all the activity logs.

i.e. `Promise<import('@prisma/client').DocumentActivityLog>[]`